MobileCenter/MobileCenter/Internals/AppDelegate/MSAppDelegateUtil.h

```diff
@@ -0,0 +1,18 @@
+#if TARGET_OS_OSX
+#import <AppKit/AppKit.h>
+#ifndef MSOriginalAppDelegate
+#define MSOriginalAppDelegate NSApplicationDelegate
+#endif
+#ifndef MSOriginalApplication
```

Comment:
We should check MSOriginalAppDelegate and define all of them if it is not defined.
Even though UIApplication and NSApplication cannot exist together, we have to make sure Xcode can also give us an error (by warnings as errors) for redefining issue.

Reply:
Actually `MSOriginalApplication` should be defined in the utility class since it's not only used by the app delegate forwarder.

Reply:
If that's the case, you can check the definition of Application then just define AppDelegate based on platform?

Reply:
We'll play it safe and keep it as it is.